File: 1/2021.11.04/2054.两个最好的不重叠活动/java/solution.java
class Solution {
    public int maxTwoEvents(int[][] events) {
        List<Event> list = new ArrayList<>();
        for (int[] event : events) {
            list.add(new Event(event[0], 0, event[2]));
            list.add(new Event(event[1], 1, event[2]));
        }
        Event[] arr = list.toArray(new Event[0]);
        Arrays.sort(arr, (a, b) -> a.ts == b.ts ? a.op - b.op : a.ts - b.ts);
        int res = 0, best = 0;
        for (Event event : arr) {
            if (event.op == 0) {
                res = Math.max(res, best + event.val);
            } else {
                best = Math.max(best, event.val);
            }
        }
        return res;
    }
}

class Event {
    public int ts;
    public int op;
    public int val;

    public Event(int ts, int op, int val) {
        this.ts = ts;
        this.op = op;
        this.val = val;
    }
}

File: 1/2021.11.04/2054.两个最好的不重叠活动/python/solution.py
class Solution:
    def maxTwoEvents(self, events: List[List[int]]) -> int:
        arr = []
        for event in events:
            arr.append((event[0], 0, event[2]))
            arr.append((event[1], 1, event[2]))
        arr.sort()
        res = best = 0
        for ts, op, val in arr:
            if op == 0:
                res = max(res, best + val)
            else:
                best = max(best, val)
        return res
